### PR TITLE
add ability to specify client identifiers in rate limit rules

### PR DESCRIPTION
**Pull request by shawnps**
This adds the ability to specify client identifiers in rate limit rules, which fixes https://github.com/signalsciences/terraform-provider-sigsci/issues/60.

In order to have nested client identifiers within the `rate_limit` clause I had to change `rate_limit` from a `TypeMap` to a `TypeSet`. This means `rate_limit` will change from (example):

```
rate_limit = {
  threshold = 6 
  interval  = 10
  duration  = 300 
}
```

to (example):

```
rate_limit {
  threshold = 6 
  interval  = 10
  duration  = 300 

  client_identifiers {
    type = "ip"
  }
}
```

I'm also making `client_identifiers` a required attribute because while the API will default to 
```
client_identifiers {
    type = "ip"
}
```

when there is no `client_identifiers` specified, any `terraform plan` after the initial `apply` will show a diff in `client_identifiers` if it's not in the TF.

### CODE REVIEW

**Comment by tdysko-cf** [comment]
hi, any updates on this?

**Comment by nicorahm99** [comment]
Hello, are there any updates on this request. We would appreciate this functionality as there is no possibility to use the ratelimits of Sig-Sci as intended without identifying a single client.

File: docs/guides/3.0.0.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+## Upgrading to version 3
+
+**Rate limit rules now support non-IP client identifiers**
+
+To add support for non-IP client identifiers within rate limit rules w ehad to convert them from `TypeMap` to `TypeSet`.
```
Comment (shawnps) [inline docs/guides/3.0.0.md:5]:
```suggestion
To add support for non-IP client identifiers within rate limit rules we had to convert them from `TypeMap` to `TypeSet`.
```

**Review by daniel-corbett — APPROVED** [review]

**Comment by daniel-corbett** [comment]
Hi all, we have officially released[ version 3.0.0 of the NGWAF Terraform provider](https://registry.terraform.io/providers/signalsciences/sigsci/latest). This adds support for non-IP client identifiers.
This is a breaking change. Please refer to the upgrade guide, which outlines the changes that you will need to make in order to upgrade:
https://registry.terraform.io/providers/signalsciences/sigsci/latest/docs/guides/3.0.0

cc @sjparkinson @tdysko-cf @nicorahm99

**Pull request merged**